contracts/polyx-vesting/lib.rs

```
@@ -0,0 +1,168 @@
+                Self::new_init(
+                    contract,
+                    beneficiary_address,
+                    start_timestamp,
+                    duration_seconds,
+                )
+            })
+        }
+
+        fn new_init(
+            &mut self,
+            beneficiary_address: AccountId,
+            start_timestamp: u128,
+            duration_seconds: u128,
+        ) {
+            self.beneficiary = beneficiary_address;
+            self.start = start_timestamp;
+            self.duration = duration_seconds;
+        }
+
+        // Getters
+        /// Returns the vesting duration.
+        #[ink(message)]
+        pub fn duration(&self) -> u128 {
+            self.duration
+        }
+
+        /// Returns the start timestamp.
+        #[ink(message)]
+        pub fn start(&self) -> u128 {
+            self.start
+        }
+
+        /// Returns the beneficiary address.
+        #[ink(message)]
+        pub fn beneficiary(&self) -> AccountId {
+            self.beneficiary
+        }
+
+        /// Returns the amount of POLYX already released.
+        #[ink(message)]
+        pub fn released(&self) -> Balance {
+            self.released
+        }
+
+        /// Returns the amount of releasable POLYX.
+        #[ink(message)]
+        pub fn releasable(&self) -> Balance {
+            self.vested_amount(self.env().block_timestamp().into())
+                .saturating_sub(self.released())
+        }
+
+        #[ink(message)]
+        pub fn get_time(&self) -> Balance {
+            self.env().block_timestamp().into()
+        }
+
+        #[ink(message)]
+        pub fn get_balance(&self) -> Balance {
+            self.env().balance()
+        }
+
+        /// Release the native token (POLYX) that have already vested.
+        #[ink(message)]
+        pub fn release(&mut self) -> Result<()> {
+            let amount = self.releasable();
+            assert!(amount > 0, "insufficient funds!");
```

Should return an error here to tell the caller that funds are not releasable yet.